cli/initiator/initiator.go

```diff
@@ -29,138 +30,209 @@ func init() {
+			logger.Fatal("😥 Failed to load participants: ", zap.Error(err))
+		}
+		privKeyPath := viper.GetString("initiatorPrivKey")
+		if privKeyPath == "" {
+			logger.Fatal("😥 Failed to get initiator key flag value", zap.Error(err))
+		}
+		var privateKey *rsa.PrivateKey
+		pass := viper.GetString("initiatorPrivKeyPassword")
+		if pass != "" {
+			// check if a password string a valid path, then read password from the file
+			if _, err := os.Stat(pass); err != nil {
+				logger.Fatal("😥 Password file: ", zap.Error(err))
+			}
+			keyStorePassword, err := os.ReadFile(pass)
+			if err != nil {
+				logger.Fatal("😥 Error reading password file: ", zap.Error(err))
+			}
+			encryptedJSON, err := os.ReadFile(privKeyPath)
+			if err != nil {
+				logger.Fatal("😥 Cant read operator`s key file", zap.Error(err))
+			}
+			privateKey, err = crypto.ConvertEncryptedPemToPrivateKey(encryptedJSON, string(keyStorePassword))
+			if err != nil {
+				logger.Fatal(err.Error())
+			}
+		} else {
+			privateKey, err = crypto.PrivateKey(privKeyPath)
+			if err != nil {
+				logger.Fatal(err.Error())
+			}
 		}
-		dkgClient := client.New(opMap)
 
+		dkgInitiator := initiator.New(privateKey, opMap, logger)
 		withdrawAddr := viper.GetString("withdrawAddress")
 		if withdrawAddr == "" {
-			logger.Fatal("failed to get withdrawal address flag value", zap.Error(err))
+			logger.Fatal("😥 Failed to get withdrawal address flag value: ", zap.Error(err))
 		}
-		forkHex := viper.GetString("fork")
-		if forkHex == "" {
-			logger.Fatal("failed to get fork version flag value", zap.Error(err))
+		fork := viper.GetString("fork")
+		if fork == "" {
+			logger.Fatal("😥 Failed to get fork version flag value: ", zap.Error(err))
 		}
-		if err != nil {
-			logger.Fatal("failed to get fork version flag value", zap.Error(err))
-		}
-		forkBytes, err := hex.DecodeString(forkHex)
-		if err != nil {
-			logger.Fatal("failed to get fork version flag value", zap.Error(err))
-		}
-		var fork [4]byte
-		copy(fork[:], forkBytes)
-		var forkName string
+		var forkHEX [4]byte
 		switch fork {
-		case [4]byte{0x00, 0x00, 0x10, 0x20}:
-			forkName = "prater"
-		case [4]byte{0, 0, 0, 0}:
-			forkName = "mainnet"
-		case [4]byte{0x99, 0x99, 0x99, 0x99}:
-			forkName = "now_test_network"
+		case "prater":
+			forkHEX = [4]byte{0x00, 0x00, 0x10, 0x20}
+		case "mainnet":
+			forkHEX = [4]byte{0, 0, 0, 0}
+		case "now_test_network":
+			forkHEX = [4]byte{0x99, 0x99, 0x99, 0x99}
 		default:
-			forkName = "mainnet"
+			logger.Fatal("😥 Please provide a valid fork name: mainnet, prater, or now_test_network")
 		}
 		owner := viper.GetString("owner")
 		if owner == "" {
-			logger.Fatal("failed to get owner address flag value", zap.Error(err))
+			logger.Fatal("😥 Failed to get owner address flag value: ", zap.Error(err))
 		}
 		nonce := viper.GetUint64("nonce")
-		withdrawPubKey, err := hex.DecodeString(withdrawAddr)
+		withdrawPubKey := common.HexToAddress(withdrawAddr).Bytes()
```

Comment:
This is dangerous for user input because it fails silently, so you can pass just a "0" and this would not fail.

WDYT about making our own parsing func that would return an error if it's too long, too short or not a valid hex?

Reply:
added HexToAddress with error handling to utils + unit tests for it